features/telemetry/daemon/src/main/java/org/opennms/netmgt/telemetry/daemon/TelemetryMessageConsumer.java

```diff
@@ -99,34 +106,59 @@ public void handleMessage(TelemetryProtos.TelemetryMessageLog messageLog) {
     }
 
     private Adapter buildAdapter(org.opennms.netmgt.telemetry.config.model.Adapter adapterDef) throws Exception {
-        // Instantiate the associated class
-        final Object adapterInstance;
-        try {
-            final Class<?> clazz = Class.forName(adapterDef.getClassName());
-            final Constructor<?> ctor = clazz.getConstructor();
-            adapterInstance = ctor.newInstance();
-        } catch (Exception e) {
-            throw new RuntimeException(String.format("Failed to instantiate adapter with class name '%s'.",
-                    adapterDef.getClassName(), e));
-        }
 
-        // Cast
-        if (!(adapterInstance instanceof Adapter)) {
-            throw new IllegalArgumentException(String.format("%s must implement %s", adapterDef.getClassName(), Adapter.class.getCanonicalName()));
+        AdapterFactory adapterFactory = null;
+        while (ManagementFactory.getRuntimeMXBean().getUptime() < GRACE_PERIOD_MS) {
```

Comment:
If the daemon is reloaded after the system has been up, we won't even attempt a lookup in the registry.